tofd/pars/R3BTofDCal2HitPar.cxx

```diff
@@ -297,1099 +352,2668 @@ void R3BTofDCal2HitPar::Exec(Option_t* option)
-                    if (!para)
-                    {
-                        R3BLOG(error,
-                               "Hit par not found, Plane: " << top->GetDetectorId() << ", Bar: " << top->GetBarId());
-                        continue;
-                    }
-
-                    // calculate tdiff with offest
-                    auto tdiff = (bot_ns + para->GetOffset1()) - (top_ns + para->GetOffset2());
-                    /*
-                                        // walk corrections
-                                        if (para->GetPar1Walk() == 0. || para->GetPar2Walk() == 0. ||
-                       para->GetPar3Walk() == 0. || para->GetPar4Walk() == 0. || para->GetPar5Walk() == 0.){
-                                            R3BLOG(warn, "Walk correction not found!");
-                                        }else{
-
-                                        bot_ns = bot_ns - walk(bot_tot,
-                                                               para->GetPar1Walk(),
-                                                               para->GetPar2Walk(),
-                                                               para->GetPar3Walk(),
-                                                               para->GetPar4Walk(),
-                                                               para->GetPar5Walk());
-                                        top_ns = top_ns - walk(top_tot,
-                                                               para->GetPar1Walk(),
-                                                               para->GetPar2Walk(),
-                                                               para->GetPar3Walk(),
-                                                               para->GetPar4Walk(),
-                                                               para->GetPar5Walk());
-                                        }*/
-
-                    auto posToT =
-                        para->GetLambda() * log((top_tot + para->GetToTOffset2()) / (bot_tot + para->GetToTOffset1()));
-
-                    // fill control histograms
-                    // fhTot1vsTot2[iPlane - 1][iBar - 1]->Fill(top_tot, bot_tot);
-                    fhSqrtQvsPosToT[iPlane - 1][iBar - 1]->Fill(posToT, sqrt(top_tot * bot_tot));
-
-                    // Time differences of one paddle
-                    fhTdiff[iPlane - 1]->Fill(iBar, tdiff);
-
-                    // Time of hit
-                    auto THit = (bot_ns + top_ns) / 2. - para->GetSync();
-
-                    // Sync of one plane
-                    fhTsync[iPlane - 1]->Fill(iBar, THit);
-                }
-
-                // prepare double exponential fit
-                if (!fTofdSmiley && fTofdQ > 0.1 && fParameter > 1)
-                {
-                    LOG(debug) << "Prepare histo for double exponential fit";
-                    auto par = fHitPar->GetModuleParAt(iPlane, iBar);
```

Comment:
Why not make !TofdSmiley and fTofdQ>0.1 a consition within (fTofd > 0 && fParameter > 1)?

Why do you need the fTofdQ>0.1 requirement? Would !TofdSmiley be sufficient?